include/oneapi/dpl/async

```diff
@@ -19,13 +19,26 @@
 #include "oneapi/dpl/internal/common_config.h"
 #include "oneapi/dpl/pstl/onedpl_config.h"
 
-#include "oneapi/dpl/internal/async_extension_defs.h"
+#if _ONEDPL_BACKEND_SYCL
+#    if !_ONEDPL_ASYNC_FORWARD_DECLARED
+#        include "oneapi/dpl/internal/async_extension_defs.h"
+#        define _ONEDPL_ASYNC_FORWARD_DECLARED 1
+#    endif // !_ONEDPL_ASYNC_FORWARD_DECLARED
 
-#if _ONEDPL_EXECUTION_POLICIES_DEFINED
+#    if _ONEDPL_EXECUTION_POLICIES_DEFINED
 // If <execution> has already been included, pull in implementations
-#    include "oneapi/dpl/internal/async_impl/async_impl.h"
-#    include "oneapi/dpl/internal/async_impl/glue_async_impl.h"
-#endif // _ONEDPL_EXECUTION_POLICIES_DEFINED
+#        include "oneapi/dpl/internal/async_impl/async_utils.h"
+#        include "oneapi/dpl/internal/async_impl/async_impl.h"
+#        include "oneapi/dpl/internal/async_impl/glue_async_impl.h"
+#    endif // _ONEDPL_EXECUTION_POLICIES_DEFINED
+#endif     // _ONEDPL_BACKEND_SYCL
+
+namespace oneapi
+{
+namespace dpl
+{
+}; // namespace dpl
+}; // namespace oneapi
 
 namespace dpl = oneapi::dpl;
```

Comment:
From my point of view - we are unable to move this empty namespace definitions under ```#if _ONEDPL_BACKEND_SYCL``` - as I understand we should have declared ```dpl``` namespace after include of this header regardless of ```_ONEDPL_BACKEND_SYC``` state.

Reply:
From my point of view - we should still have this declaration to avoid the compile error in our new tests
in case when we use ```-DONEDPL_BACKEND=tbb``` :
```
FAILED: test/CMakeFiles/header_inclusion_order_async_1.pass.dir/general/header_inclusion_order_async_1.pass.cpp.o 
/usr/bin/oneapi_compiler/linux_prod/compiler/linux/bin/icpx -DONEDPL_USE_DPCPP_BACKEND=0 -DPSTL_USE_DEBUG -DTBB_USE_DEBUG=1 -D_PSTL_TEST_SUCCESSFUL_KEYWORD=1 -I/oneDPL/test -I/oneDPL/include -isystem /usr/lib/oneapi-tbb-2021.5.0/include -fsycl -O0 -g -fno-fast-math -fopenmp-simd -MD -MT test/CMakeFiles/header_inclusion_order_async_1.pass.dir/general/header_inclusion_order_async_1.pass.cpp.o -MF test/CMakeFiles/header_inclusion_order_async_1.pass.dir/general/header_inclusion_order_async_1.pass.cpp.o.d -o test/CMakeFiles/header_inclusion_order_async_1.pass.dir/general/header_inclusion_order_async_1.pass.cpp.o -c /oneDPL/test/general/header_inclusion_order_async_1.pass.cpp
In file included from /oneDPL/test/general/header_inclusion_order_async_1.pass.cpp:18:
/oneDPL/include/oneapi/dpl/async:34:17: error: use of undeclared identifier 'oneapi'
namespace dpl = oneapi::dpl;
                ^
1 error generated.
```

Reply:
There should not be such an issue with what I have proposed before:

> Can't we just move `namespace dpl = oneapi::dpl;` to the scope of `#if _ONEDPL_BACKEND_SYCL` ... ?

Reply:
Fixed.